sort.c

```diff
@@ -10,30 +10,52 @@
 
 #include "redisraft.h"
 
-static void replySortedArray(RedisModuleCtx *ctx, RedisModuleCallReply * reply)
+static void replySortedArray(RedisModuleCtx *ctx, RedisModuleCallReply * reply, int reply_type)
 {
     size_t len = RedisModule_CallReplyLength(reply);
-    if (len == 0) {
-        RedisModule_ReplyWithNullArray(ctx);
-        return;
-    }
 
-    RedisModuleDict * dict = RedisModule_CreateDict(ctx);
-    for(int i=0; i < len; i++) {
-        RedisModuleCallReply * entry = RedisModule_CallReplyArrayElement(reply, i);
+    RedisModuleDict *dict = RedisModule_CreateDict(ctx);
+    for(size_t i = 0; i < len; i++) {
+        long *val;
+        RedisModuleCallReply * entry;
+        switch (reply_type) {
+            case REDISMODULE_REPLY_ARRAY:
+                entry = RedisModule_CallReplyArrayElement(reply, i);
+                break;
+            case REDISMODULE_REPLY_SET:
+                entry = RedisModule_CallReplySetElement(reply, i);
+                break;
+        }
+
         size_t entry_len;
         const char * entry_str = RedisModule_CallReplyStringPtr(entry, &entry_len);
-        RedisModule_DictSetC(dict, (char *) entry_str, entry_len, NULL);
+        val = RedisModule_DictGetC(dict, (char *) entry_str, entry_len, NULL);
+        if (!val) {
```

Comment:
How about just casting the `void *` to `unsigned long` to keep this counter? A bit hackish but will be much more efficient.

Reply:
I dont understand this.

Reply:
Instead of allocating a `long`, storing a pointer to it in the dict and incrementing the value the pointer points to - do something like:

```c
unsigned long myval = (unsigned long) RM_DictGetC(...);
myval++;
RM_DictReplaceC(..., (void *) myval);
```